File: vertx-lang-kotlin-coroutines-codegen/src/main/java/io/vertx/kotlin/coroutines/KotlinCoroutineGeneratorLoader.java
package io.vertx.kotlin.coroutines;

import io.vertx.codegen.Generator;
import io.vertx.codegen.GeneratorLoader;

import javax.annotation.processing.ProcessingEnvironment;
import java.util.stream.Stream;

public class KotlinCoroutineGeneratorLoader implements GeneratorLoader {
  @Override
  public Stream<Generator<?>> loadGenerators(ProcessingEnvironment processingEnvironment) {
    return Stream.of(new KotlinCoroutineGenerator());
  }
}

File: vertx-lang-kotlin-coroutines-codegen/src/main/kotlin/io/vertx/kotlin/coroutines/KotlinCoroutineGeneratorLoader.kt
package io.vertx.kotlin.coroutines

import io.vertx.codegen.Generator
import io.vertx.codegen.GeneratorLoader
import java.util.stream.Stream
import javax.annotation.processing.ProcessingEnvironment

class KotlinCoroutineGeneratorLoader : GeneratorLoader {
  override fun loadGenerators(processingEnv: ProcessingEnvironment?): Stream<Generator<*>> {
    return Stream.of(KotlinCoroutineGenerator())
  }
}
